lib/buffer.js

```diff
@@ -54,17 +54,19 @@ function alignPool() {
   }
 }
 
+const bufferDeprecationWarning =
+    deprecate(() => {}, 'Buffer constructor is deprecated. ' +
+      'Use Buffer.from, Buffer.allocUnsafe or Buffer.alloc instead.');
+
 /**
- * The Buffer() construtor is "soft deprecated" ... that is, it is deprecated
- * in the documentation and should not be used moving forward. Rather,
- * developers should use one of the three new factory APIs: Buffer.from(),
- * Buffer.allocUnsafe() or Buffer.alloc() based on their specific needs. There
- * is no hard deprecation because of the extent to which the Buffer constructor
- * is used in the ecosystem currently -- a hard deprecation would introduce too
- * much breakage at this time. It's not likely that the Buffer constructors
- * would ever actually be removed.
+ * The Buffer() construtor is deprecated ... that is, it should not be used
+ * moving forward. Rather, developers should use one of the three new factory
+ * APIs: Buffer.from(), Buffer.allocUnsafe() or Buffer.alloc() based on their
+ * specific needs. It's not likely that the Buffer constructors would ever
+ * actually be removed.
  **/
 function Buffer(arg, encodingOrOffset, length) {
+  bufferDeprecationWarning();
```

Comment:
Hmm, these are some valid points. I am actually quite sure there’s no reason to change the semantics of `new Buffer()`, with all combinations of arguments, but I should probably write a proof of concept for that before making definitive statements. ;)